src/tz_android.rs

```diff
@@ -1,16 +1,27 @@
 use std::ffi::CStr;
+use std::sync::Once;
 
 use android_system_properties::AndroidSystemProperties;
-use once_cell::sync::OnceCell;
+
+static INITALIZED: Once = Once::new();
+static mut PROPERTIES: Option<AndroidSystemProperties> = None;
 
 // From https://android.googlesource.com/platform/ndk/+/android-4.2.2_r1.2/docs/system/libc/OVERVIEW.html
 // The system property named 'persist.sys.timezone' contains the name of the current timezone.
-
-static PROPERTIES: OnceCell<AndroidSystemProperties> = OnceCell::new();
+// SAFETY: the key is NUL-terminated and there are no other NULs
+const KEY: &CStr = unsafe { CStr::from_bytes_with_nul_unchecked(b"persist.sys.timezone\0") };
```

Comment:
`CStr::from_bytes_with_nul_unchecked` is only usable in const contexts since Rust 1.59.0, so this PR actually raises the MSRV from where `once_cell` would have bumped it to.

https://doc.rust-lang.org/nightly/std/ffi/struct.CStr.html#method.from_bytes_with_nul_unchecked

Reply:
Actually, that would be in line with our MSRV policy not to care about tier 2 targets. :) The MSRV for tier 1 targets was still reduced.

But yes, rising it for Android was an accident. Do you want to open a PR to fix that?

Reply:
I would rather open a PR to revert this change.